demos/generomak_solps_from_raw_files.py

```diff
@@ -0,0 +1,234 @@
+c6_samples = sample3d_grid(c6.distribution.density, xsamp, [0], zsamp).squeeze()
+# Magnitude of main ion velocity vector.
+h1_velocity = samplevector3d_grid(h1.distribution.bulk_velocity, xsamp, [0], zsamp).squeeze()
+h1_speed = linalg.norm(h1_velocity, axis=-1)
+# Mask determining whether a point is inside or outside the simulation mesh.
+# This is a 2D function: see sim.inside_volume_mesh for the 3D equivalent.
+inside_samples = sample2d_grid(sim.inside_mesh, xsamp, zsamp)
+
+# Plot the simulation mesh.
+plt.ion()
+mesh.plot_mesh()
+plt.title('Mesh geometry')
+
+# Plot sampled values.
+plt.figure()
+plt.imshow(ne_samples.T, extent=[xl, xu, zl, zu], origin='lower', norm=LogNorm())
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('Electron density [m-3]')
+
+plt.figure()
+plt.imshow(te_samples.T, extent=[xl, xu, zl, zu], origin='lower')
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('Electron temperature [eV]')
+
+plt.figure()
+plt.imshow(h0_samples.T, extent=[xl, xu, zl, zu], origin='lower', norm=LogNorm())
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('H0 density [m-3]')
+
+plt.figure()
+plt.imshow(h1_samples.T, extent=[xl, xu, zl, zu], origin='lower', norm=LogNorm())
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('H+ density [m-3]')
+
+plt.figure()
+plt.imshow(c0_samples.T, extent=[xl, xu, zl, zu], origin='lower', norm=LogNorm())
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('CI density [m-3]')
+
+plt.figure()
+plt.imshow(c1_samples.T, extent=[xl, xu, zl, zu], origin='lower', norm=LogNorm())
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('CII density [m-3]')
+
+plt.figure()
+plt.imshow(c2_samples.T, extent=[xl, xu, zl, zu], origin='lower', norm=LogNorm())
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('CIII density [m-3]')
+
+plt.figure()
+plt.imshow(c3_samples.T, extent=[xl, xu, zl, zu], origin='lower', norm=LogNorm())
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('CIV density [m-3]')
+
+plt.figure()
+plt.imshow(c4_samples.T, extent=[xl, xu, zl, zu], origin='lower', norm=LogNorm())
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('CV density [m-3]')
+
+plt.figure()
+plt.imshow(c5_samples.T, extent=[xl, xu, zl, zu], origin='lower', norm=LogNorm())
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('CVI density [m-3]')
+
+plt.figure()
+plt.imshow(c6_samples.T, extent=[xl, xu, zl, zu], origin='lower', norm=LogNorm())
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('CVII density [m-3]')
+
+plt.figure()
+plt.imshow(h1_speed.T, extent=[xl, xu, zl, zu], origin='lower')
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('H+ speed [m/s]')
+
+plt.figure()
+plt.imshow(inside_samples.T, extent=[xl, xu, zl, zu], origin='lower')
+plt.colorbar()
+plt.xlim(xl, xu)
+plt.ylim(zl, zu)
+plt.title('Inside/Outside test')
+
+input('Press Return to continue...')
+plt.close('all')
```

Comment:
Is the input and closing really needed? If you remove this the user could be already enjoying the profile plots while the observation is running.

Reply:
I put it in because every 15 seconds when the observation updates the plot it shifts focus to the camera plot. I decided to wait until the user is done admiring the profiles before starting the observation. But the suggestion to separate the demo into two is a good one which I'll implement, so closing the plots will become unnecessary.